### PR TITLE
Fix README links

### PR DESCRIPTION
**Pull request by TheJavaGuy**
This PR fixes links for all languages that point to Collections website. It also reorders them so that the order is the same as in the website's Language drop-down menu. Additionally, it rewrites old www.eclipse.org/collections URLs with new eclipse.dev ones.

@donraab @motlin @Desislav-Petrov @nikhilnanivadekar when you have a bit of time please review :)

### CODE REVIEW

**Review by donraab — APPROVED** [review]
LGTM. @nikhilnanivadekar will wait for your review/approval to merge.

**Comment by nikhilnanivadekar** [comment]
Can you please squash in to a single commit? And rebase on master? Thanks!

**Comment by TheJavaGuy** [comment]
@nikhilnanivadekar Done and done ✅

**Review by nikhilnanivadekar — APPROVED** [review]

**Pull request merged**
